### PR TITLE
[ROCm] Enable inductor GEMM lowering for gfx11

### PR DESCRIPTION
**Pull request by iupaikov-amd**
This check doesn't make sense for some of the AMD gpus since they have the right amount of CUs but multi_processor_count returns WGPs on RDNA while still performing adequately. A lot of tests fail on modern archs due to this check defaulting them to not using the GEMMs backend.

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @jataylo @hongxiayang @naromero77amd @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

### CODE REVIEW

**Review by pruthvistony — APPROVED** [review]

File: torch/_inductor/utils.py [hunk 1]
```diff
@@ -1104,8 +1104,20 @@ def _new_line(self, line: str) -> DelayReplaceLine:
 
 @functools.lru_cache(None)
 def is_big_gpu(index) -> bool:
+    prop = torch.cuda.get_device_properties(index)
+
+    # SM logic is not relevant to ROCm gpus
+    # Arbitrarily skipping the older models
+    if torch.version.hip is not None:
```
Comment (jeffdaily) [inline torch/_inductor/utils.py:1111]:
```suggestion
    if torch.version.hip:
```
nit.  the above suggestion is more succinct.

**Review by jeffdaily — APPROVED** [review]

**Comment by iupaikov-amd** [comment]
@jeffdaily added your nitpick, can you approve the workflows again, please?

**Review by malfet — APPROVED** [review]

**Comment by iupaikov-amd** [comment]
@malfet applied the linter formatting, thought I'm not sure this is what caused merge to fail

**Pull request closed without merging**